plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -213,7 +227,13 @@ public ConnectorPageSource createPageSource(
                         .requiredSchema(),
                 typeManager);
 
-        Map<Integer, Optional<String>> partitionKeys = split.getPartitionKeys();
+        PartitionSpec partitionSpec = PartitionSpecParser.fromJson(tableSchema, table.getPartitionSpecJson());
```

Comment:
This block is not right, breaks partition evolution. Looking into another solution